vespa-feed-client/src/main/java/ai/vespa/feed/client/impl/HttpFeedClient.java

```diff
@@ -56,18 +57,20 @@ class HttpFeedClient implements FeedClient {
     private final boolean speedTest;
 
     HttpFeedClient(FeedClientBuilderImpl builder) throws IOException {
-        this(builder, builder.dryrun ? new DryrunCluster() : new JettyCluster(builder));
+        this(builder,
+             builder.dryrun ? () -> new DryrunCluster()
+                            : () -> { try { return new JettyCluster(builder); } catch (IOException e) { throw new UncheckedIOException(e); } });
```

Comment:
Consider adding a `ClusterFactory` interface and using `clusterFactory` instead of `clusters` as variable name to clarify its intention.